src/contracts/oracles/ChainlinkRelayer.sol

```diff
@@ -29,53 +31,64 @@ contract ChainlinkRelayer is IBaseOracle, IChainlinkRelayer {
     staleThreshold = _staleThreshold;
-    chainlinkFeed = IChainlinkOracle(_aggregator);
 
-    multiplier = 18 - chainlinkFeed.decimals();
-    symbol = chainlinkFeed.description();
+    multiplier = 18 - priceFeed.decimals();
+    symbol = priceFeed.description();
   }
 
   /// @inheritdoc IBaseOracle
   function getResultWithValidity() external view returns (uint256 _result, bool _validity) {
     // Fetch values from Chainlink
-    (, int256 _aggregatorResult,, uint256 _aggregatorTimestamp,) = chainlinkFeed.latestRoundData();
+    (, int256 _feedResult,, uint256 _feedTimestamp,) = priceFeed.latestRoundData();
 
     // Parse the quote into 18 decimals format
-    _result = _parseResult(_aggregatorResult);
+    _result = _parseResult(_feedResult);
 
     // Check if the price is valid
-    _validity = _aggregatorResult > 0 && _isValidFeed(_aggregatorTimestamp);
+    _validity = _feedResult > 0 && _isValidFeed(_feedTimestamp);
   }
 
   /// @inheritdoc IBaseOracle
   function read() external view returns (uint256 _result) {
     // Fetch values from Chainlink
-    (, int256 _aggregatorResult,, uint256 _aggregatorTimestamp,) = chainlinkFeed.latestRoundData();
+    (, int256 _feedResult,, uint256 _feedTimestamp,) = priceFeed.latestRoundData();
 
     // Revert if price is invalid
-    if (_aggregatorResult <= 0 || !_isValidFeed(_aggregatorTimestamp)) revert InvalidPriceFeed();
+    if (_feedResult <= 0 || !_isValidFeed(_feedTimestamp)) revert InvalidPriceFeed();
 
     // Parse the quote into 18 decimals format
-    _result = _parseResult(_aggregatorResult);
+    _result = _parseResult(_feedResult);
   }
 
-  /// @notice Parses the result from the aggregator into 18 decimals format
-  function _parseResult(int256 _chainlinkResult) internal view returns (uint256 _result) {
-    return uint256(_chainlinkResult) * 10 ** multiplier;
+  /// @notice Parses the result from the price feed into 18 decimals format
+  function _parseResult(int256 _feedResult) internal view returns (uint256 _result) {
+    return uint256(_feedResult) * 10 ** multiplier;
   }
 
-  /// @notice Checks if the feed is valid, considering the staleThreshold and the feed timestamp
+  /// @notice Checks if the feed is valid, considering the sequencer status, the staleThreshold and the feed timestamp
   function _isValidFeed(uint256 _feedTimestamp) internal view returns (bool _valid) {
-    uint256 _now = block.timestamp;
-    if (_feedTimestamp > _now) return false;
-    return _now - _feedTimestamp <= staleThreshold;
+    // Check the sequencer status
+    (, int256 _feedStatus,,,) = sequencerUptimeFeed.latestRoundData();
+
+    // Status == 0: Sequencer is up
+    // Status == 1: Sequencer is down
+    bool _isSequencerUp = _feedStatus == 0;
+    if (!_isSequencerUp) return false;
+
+    // Make sure the staleThreshold has not passed after the feed timestamp
+    uint256 _timeSinceFeed = block.timestamp - _feedTimestamp;
+    return _timeSinceFeed <= staleThreshold;
```

Comment:
Should a sequencer uptime grace period check be included during feed validation?
```solidity
uint256 internal constant _GRACE_PERIOD_TIME = 3600;

// Check the sequencer status
(, int256 _feedStatus, uint256 _startedAt,,) = sequencerUptimeFeed.latestRoundData();

// Make sure the grace period has passed after the sequencer is back up
uint256 _timeSinceUp = block.timestamp - _startedAt;
if (_timeSinceUp <= _GRACE_PERIOD_TIME) return false;
```